src/distilabel/pipeline/_dag.py

```diff
@@ -307,3 +309,52 @@ def _validate_step_process_runtime_parameters(self, step: "Step") -> None:
                     f"Step '{step.name}' is missing required runtime parameter '{param_name}'."
                     " Please, provide a value for it when calling `Pipeline.run`"
                 )
+
+    def _model_dump(self, obj: Any, **kwargs: Any) -> Dict[str, Any]:
+        """Dumps the content of the DAG to a dict.
+
+        References:
+        https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.json_graph.adjacency_data.html#networkx.readwrite.json_graph.adjacency_data
```

Comment:
That's the same formatting we were applying before just so that's more readable in the docs

```suggestion
        References:
        * [`adjacency_data` - NetworkX Documentation](https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.json_graph.adjacency_data.html#networkx.readwrite.json_graph.adjacency_data)
```